File: src/app/android/app/src/main/java/com/cyberacy/app/models/entities/Person.kt
package com.cyberacy.app.models.entities

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

class Person(
    @SerializedName("nir")
    val nir: String,

    @SerializedName("firstname")
    val firstname: String,

    @SerializedName("lastname")
    val lastname: String,

    @SerializedName("email")
    val email: String,

    @SerializedName("birthday")
    val birthday: String?,

    @SerializedName("address_street")
    val address_street: String,

    @SerializedName("town")
    val town: String?,

    @SerializedName("town_code_insee")
    val townCodeInsee: String,

    @SerializedName("sex")
    val sex: Int,

    @SerializedName("profile")
    val profile: Int,

    @SerializedName("password")
    val password: String?,

    @SerializedName("token")
    val token: String?,
) { }

File: src/client/flutter/lib/models/entities/person.dart
class Person {
  String nir;
  String firstname;
  String lastname;
  String? email;
  DateTime? birthday;
  String? addressStreet;
  String? town;
  int sex;
  int? profile;
  String? token;

  Person({
    required this.nir,
    required this.firstname,
    required this.lastname,
    this.email,
    this.birthday,
    this.addressStreet,
    this.town,
    required this.sex,
    this.profile,
    this.token,
  });

  Person.fromJson(Map<String, dynamic> json)
      : nir = json["nir"],
        firstname = json["firstname"],
        lastname = json["lastname"],
        email = json["email"],
        birthday = DateTime.parse(json["birthday"]),
        addressStreet = json["address_street"],
        town = json["town"],
        sex = json["sex"],
        profile = json["profile"],
        token = json["token"];


}
